homeassistant/components/tplink/strings.json

```diff
@@ -95,6 +95,13 @@
           "off": "[%key:component::binary_sensor::entity_component::moisture::state::off%]",
           "on": "[%key:component::binary_sensor::entity_component::moisture::state::on%]"
         }
+      },
+      "motion_detected": {
+        "name": "[%key:component::binary_sensor::entity_component::motion::name%]",
+        "state": {
+          "off": "[%key:component::binary_sensor::entity_component::gas::state::off%]",
+          "on": "[%key:component::binary_sensor::entity_component::gas::state::on%]"
```

Comment:
I don't really like this c&p from `entity_component::motion` as I'd like to trust `binary_sensor` maintainers' to avoid out-of-sync here, but hassfest doesn't seem to like indirection so..

Reply:
We took a decision to enforce translation strings for all of our entities to avoid missing translation keys that sometimes were able to use a device class or perhaps not.  There is a test for this in the test framework to ensure that they exist.

Reply:
But can't you also assert that a device class is set or the name is equal to None?

Reply:
This is now fixed with #128022.